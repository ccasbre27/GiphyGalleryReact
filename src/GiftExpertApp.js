import React, { useState } from "react";
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GiftExpertApp = ({ defaultCategories = [] }) => {

    const [categories, setCategories] = useState(defaultCategories);    
   
    return (
        <div>
            <h2>Gift Expert App</h2>
            <hr />
            <AddCategory setCategories={ setCategories } />
            <ol>
                {
                    categories.map( category => {
                        return <GifGrid
                            key={ category } 
                            category={ category } />
                    })
                }
            </ol>
        </div>
    );
}