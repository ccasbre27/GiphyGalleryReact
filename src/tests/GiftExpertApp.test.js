import { shallow } from "enzyme";
import '@testing-library/jest-dom';
import { GiftExpertApp } from '../GiftExpertApp';

describe('Tests on <GiftExpertApp />', () => {
   
    test('should render the app', () => {
        
        const wrapper = shallow( <GiftExpertApp /> );

        expect( wrapper ).toMatchSnapshot();

    });

    test('should load a list of categories', () => {
       
        const categories = ['One Punch', 'Dragon Ball'];
        const wrapper = shallow( <GiftExpertApp defaultCategories={ categories } /> );
        
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe(categories.length);
    });
    

});
