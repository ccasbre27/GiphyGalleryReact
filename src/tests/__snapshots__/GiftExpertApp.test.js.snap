// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GiftExpertApp /> should load a list of categories 1`] = `
<div>
  <h2>
    Gift Expert App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol>
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
    <GifGrid
      category="Dragon Ball"
      key="Dragon Ball"
    />
  </ol>
</div>
`;

exports[`Tests on <GiftExpertApp /> should render the app 1`] = `
<div>
  <h2>
    Gift Expert App
  </h2>
  <hr />
  <AddCategory
    setCategories={[Function]}
  />
  <ol />
</div>
`;
